File: src/com/prettyant/s7/KtBase132.java
package com.prettyant.s7;

public class KtBase132 {
    public static void main(String[] args) {
        Person person = new Person();
//        for (String name : person.getNames()) {
//            System.out.println(name);
//        }
        for (String name : person.names) {
            System.out.println(name);
        }
    }
}

File: src/com/prettyant/s7/KtBase132.kt
package com.prettyant.s7

//TODO Kotlin语言 注解@JvmField
class Person{
    @JvmField
    val names= listOf("孙悟空","猪八戒","唐僧")
}

/*      背后代码:

         public final class Person {
            @NotNull
            private final List names = CollectionsKt.listOf(new String[]{"孙悟空", "猪八戒", "唐僧"});

            //val 只读的,只有getNames
            @NotNull
            public final List getNames() {
                return this.names;
            }

        }
            //加上 @JvmField 后,背后会提出私有方法
            //背后代码
    public final class Person {
        @JvmField
        @NotNull
        public final List names = CollectionsKt.listOf(new String[]{"孙悟空", "猪八戒", "唐僧"});
    }

 */